src/libraries/SafeERC20Metadata.sol

```diff
@@ -2,41 +2,36 @@
 pragma solidity ^0.8.0;
 
 import {IERC20Metadata} from "openzeppelin-contracts/contracts/token/ERC20/extensions/IERC20Metadata.sol";
-import {Currency, CurrencyLibrary} from "@uniswap/v4-core/src/types/Currency.sol";
 import {AddressStringUtil} from "./AddressStringUtil.sol";
 
-/// @title SafeCurrencyMetadata
+/// @title SafeERC20Metadata
 /// @notice can produce symbols and decimals from inconsistent or absent ERC20 implementations
 /// @dev Reference: https://github.com/Uniswap/solidity-lib/blob/master/contracts/libraries/SafeERC20Namer.sol
-library SafeCurrencyMetadata {
-    using CurrencyLibrary for Currency;
-
+library SafeERC20Metadata {
```

Comment:
why did you rename this back? it was currency because it also handles native right? why is it ERC20 now?

Reply:
mm i renamed it back since we're not unwrapping the currency in the file anymore, its just working with address

Reply:
mmm but its not only working with ERC20s either...? its still working with native e.g.
```
        if (addr == address(0)) {
            return nativeLabel;
        }
```

i'd personally still rather the name "currency"? its still working with both native and ERC20s even if its not using the Currency type

Reply:
`AddressRatioSortOrder` is fine.. could you `address` if you prefer. But ERC20 is still wrong imo